src/Form/AdobeAnalyticsAdminSettings.php

```diff
@@ -504,51 +599,48 @@ public function validateVariableName(array &$element, FormStateInterface $form_s
    * {@inheritdoc}
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
-    $amazon_s3_urls = [];
-    if ($form_state->getValue('mode') == 'cdn') {
-      if ($form_state->getValue('cdn_install_type') == 'amazon') {
-        $checks = [
-          'development_s_code_config' => t('Enter your development Adobe analytics tracking S Code Configuration Path.'),
-          'production_s_code_config' => t('Enter your production Adobe analytics tracking S Code Configuration Path.'),
-          'development_s_code' => t('Enter your development Adobe analytics tracking S Code Path.'),
-          'production_s_code' => t('Enter your production Adobe analytics tracking S Code Path.'),
-        ];
-        $amazon_s3_urls = [
-          'development_s_code_config',
-          'production_s_code_config',
-          'development_s_code',
-          'production_s_code',
-          'footer_js_code',
-          'cdn_custom_tracking_js_before',
-          'cdn_custom_tracking_js_after',
-        ];
-      }
-      else {
-        $checks = [
-          'development_tag_manager_container_path' => t('Enter your development Tag Manager Tool Container Path.'),
-          'production_tag_manager_container_path' => t('Enter your production Tag Manager Tool Container Path.'),
-        ];
-      }
-
-      foreach ($checks as $variable => $warning) {
-        if (empty($form_state->getValue([$variable]))) {
-          $form_state->setErrorByName($variable, $warning);
+    $config = $this->config('adobe_analytics.validation_config');
+    $cloud_fields = [
+      'development_s_code_config',
+      'production_s_code_config',
+      'development_s_code',
+      'production_s_code',
+      'development_footer_js_code',
+      'production_footer_js_code',
+      'development_cdn_custom_tracking_js_before',
+      'development_cdn_custom_tracking_js_after',
+      'production_cdn_custom_tracking_js_before',
+      'production_cdn_custom_tracking_js_after',
+      'development_tag_manager_footer_js',
+      'production_tag_manager_footer_js',
+    ];
+
+    $tag_manager_fields = [
+      'development_tag_manager_container_path',
+      'production_tag_manager_container_path',
+    ];
+
+    if ($form_state->getValue('installation_mode') == 'cdn') {
+      foreach ($cloud_fields as $field) {
+        if (!$config->get('cloud_domain') || empty($config->get('cloud_domain'))) {
+          $form_state->setErrorByName($field, "No validation criteria found. Please go to '/admin/config/adobe_analytics/validation_config' to set a validation croteria for the fields.");
```

Comment:
Please use the `Link` class to generate the Validation settings link and have the URL printed instead of the path. Also please correct `croteria` to `criteria`.